WordPress/src/main/java/org/wordpress/android/ui/reader/subfilter/SubfilterPageFragment.kt

```diff
@@ -64,65 +65,51 @@ class SubfilterPageFragment : DaggerFragment() {
 
         val category = arguments?.getSerializable(CATEGORY_KEY) as SubfilterCategory
 
+        viewModel = ViewModelProviders.of(this, viewModelFactory).get(SubfilterPageViewModel::class.java)
+        viewModel.start(category)
+
         recyclerView = view.findViewById(R.id.content_recycler_view)
         recyclerView.layoutManager = LinearLayoutManager(requireActivity())
         recyclerView.adapter = SubfilterListAdapter(uiHelpers)
 
-        viewModel = ViewModelProviders.of(requireActivity(), viewModelFactory).get(ReaderPostListViewModel::class.java)
+        emptyStateContainer = view.findViewById(R.id.empty_state_container)
+        title = emptyStateContainer.findViewById(R.id.title)
+        actionButton = emptyStateContainer.findViewById(R.id.action_button)
+
+        readerViewModel = ViewModelProviders.of(requireActivity(), viewModelFactory)
+                .get(ReaderPostListViewModel::class.java)
 
-        viewModel.subFilters.observe(this, Observer {
+        readerViewModel.subFilters.observe(this, Observer {
             (recyclerView.adapter as? SubfilterListAdapter)?.let { adapter ->
                 val items = it?.filter { it.type == category.type } ?: listOf()
+                viewModel.onManageEmptyView(items.isEmpty())
```

Comment:
Thanks for this Jirka 🙇‍♂️, I think I got the general point but not sure I got the full picture here so for now didn’t change, sorry ! I guess this means that the ReaderPostListFragment should have a reference to the SubfilterPageViewModel? Is this correct? If this applies, since the child VM is per SubfilterPageFragment this would mean to deliver (at least as it is now of course) the events to both fragments view model (probably we can combine them in some way but I would need to better think about that).

Meanwhile changed naming to `onSubfilterPageUpdated` here 89dab3793dbf1c6bdaa270ee12d007a3532b8f46

Reply:
I'd need to dive deep into the code to be sure it's even feasible. Let's not worry about it for now. Sorry for bringing it up without verifying it's feasible... I think you are right that it wouldn't be as simple as I thought.